### ISSUE TITLE
Doesn't install with GHC 7.10.2

### ISSUE THREAD

**Issue by aommm** [danr/hipspec#22]
I'm trying to install HipSpec/QuickSpec according to the instructions in the readme, but when I run `cabal install hipspec/ quickspec/` I get an error when building `structural-induction-0.1`:
```
hipspec-3.0 depends on structural-induction-0.1 which failed to install.
structural-induction-0.1 failed during the building phase. The exception was:
```

When I run it verbosely, the error with `structural-induction-0.1` is:
```
Induction/Structural/Types.hs:176:5:
    Non type-variable argument
      in the constraint: MonadState (Map (Tagged v) v') (t m)
    (Use FlexibleContexts to permit this)
    When checking that ‘f'’ has the inferred type
      f' :: forall (t :: (* -> *) -> * -> *).
            (MonadTrans t, MonadState (Map (Tagged v) v') (t m)) =>
            Tagged v -> t m v'
    In an equation for ‘unTagMapM’:
        unTagMapM f
          = flip runStateT M.empty . unTagM f'
          where
              f' tv
                = do { m <- get;
                       .... }
```

[This](http://stackoverflow.com/questions/28287329/ghc-7-10-non-type-variable-argument-use-flexiblecontexts-to-permit-this) StackOverflow thread describes how to resolve that issue, but it seems that you've already thought of this, since `structural-induction-0.3` works fine when I try to build it separately.

In conclusion: to fix this, one solution might be to upgrade HipSpec's dependency to depend on `structural-induction-0.3` instead (if this is possible).

**Comment by aommm** [comment]
Ha, now I see that this is already resolved on the `qsv2` branch. Will use that in the meantime!
This also requires QuickSpec to be on the `v2` branch.

**Comment by aommm** [comment]
Doesn't seem like it can prove anything, though. When I run `hipspec examples/Rotate.hs`, I just get the output `Proved:` and nothing else. I get the same output when I try to prove e.g. `rev xs =:= xs`.

Well, I expect that's what you get when you climb out on non-official branches. ;)

**Comment by danr** [comment]
Hi aommm,

I'm happy you want to try out hipspec. Currently, we're migrating to use our own tip library to rebuild hipspec.
In the meantime, you can use the `hipster` branch. It uses the (old) QuickSpec 0.9.6 that is on hackage. I pushed a commit now so it should find proprerties on GHC 7.10.
Do use the --auto flag to get theory exploration! Please bug us if you have further questions!
